language/use-cases/translation_training_data/translation_training_data_tsv_generator.ipynb

```diff
@@ -0,0 +1,544 @@
+    "  </td>    \n",
+    "  <td style=\"text-align: center\">\n",
+    "    <a href=\"https://console.cloud.google.com/vertex-ai/workbench/deploy-notebook?download_url=https://raw.githubusercontent.com/GoogleCloudPlatform/generative-ai/blob/main/vision/use-cases/generative-ai/language/translation/translation_training_data_generator.ipynb\">\n",
+    "      <img src=\"https://lh3.googleusercontent.com/UiNooY4LUgW_oTvpsNhPpQzsstV5W8F7rYgxgGBD85cWJoLmrOzhVs_ksK_vgx40SHs7jCqkTkCk=e14-rj-sc0xffffff-h130-w32\" alt=\"Vertex AI logo\"><br> Open in Workbench\n",
+    "    </a>\n",
+    "  </td>\n",
+    "  <td style=\"text-align: center\">\n",
+    "    <a href=\"https://github.com/GoogleCloudPlatform/generative-ai/tree/main/language/translation\">\n",
+    "      <img src=\"https://cloud.google.com/ml-engine/images/github-logo-32px.png\" alt=\"GitHub logo\"><br> View on GitHub\n",
+    "    </a>\n",
+    "  </td>\n",
+    "</table>\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "37171934-2b07-4632-9829-e7afc403f03d",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "| | |\n",
+    "|-|-|\n",
+    "|Author | [Abhijat Gupta](https://github.com/abhijat-gupta)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f163cc8e-8360-45e1-91a1-eb61f3d9e014",
+   "metadata": {},
+   "source": [
+    "## **Overview**\n",
+    "\n",
+    "[Cloud Translation API](https://cloud.google.com/translate/docs) uses Google's neural machine translation technology to let you dynamically translate text through the API using a Google pre-trained, custom model, or a translation specialized large language model (LLMs). \n",
+    "\n",
+    "It comes in [Basic and Advanced](https://cloud.google.com/translate/docs/editions) editions. Both provide fast and dynamic translation, but Advanced offers customization features, such as domain-specific translation, formatted document translation, and batch translation.\n",
+    "\n",
+    "The first 500,000 characters sent to the API to process (Basic and Advanced combined) per month are free (not applicable to LLMs).\n",
+    "\n",
+    "## Objective\n",
+    "\n",
+    "### Key Features\n",
+    "1. Paragraphs are converted into line-pairs of less than 200 words.\n",
+    "2. Tables in documents are converted into a line-pair with each row as a separate line-pair.\n",
+    "3. Limit of 200 words per line is handled.\n",
+    "4. Empty or blank lines are not added to the TSV.\n",
+    "\n",
+    "This notebook enables you to generate a TSV file out of documents(docx) for training NMT(neural machine translation) model. The generated TSV file will contain the source and target line pairs for 2 languages in 2 columns respectively. Limit of 200 words for a line is handled within the code. Example: If a line is exceeding 200 words, it won't be added to the training dataset, but will be captured and returned in a dictionary so that you can decide on how to convert it to line-pair of less than 200 words.\n",
```

Comment:
```suggestion
    "This notebook enables you to generate a TSV file out of documents (docx) for training NMT (neural machine translation) model. The generated TSV file will contain the source and target line pairs for 2 languages in 2 columns respectively. Limit of 200 words for a line is handled within the code. Example: If a line is exceeding 200 words, it won't be added to the training dataset, but will be captured and returned in a dictionary so that you can decide on how to convert it to line-pair of less than 200 words.\n",
```